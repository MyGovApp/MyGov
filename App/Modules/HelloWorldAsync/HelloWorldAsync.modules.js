import fetch from '../../Utilities/isomorphic-fetch'
import { handleAction, createReducer } from '../../Utilities/handleAction'

export const asyncToggleColor = () => {
  return dispatch => {
    dispatch({ type : 'START_REQUEST' })
    return fetch('http://192.168.0.17:3001/api/v1/helloWorldAsync')
      .then(res => res.json())
      .then(json => dispatch({ type: 'ASYNC_TOGGLE_COLOR' }))
      .catch(err => console.log('Error: ', err))
  }
}

handleAction('START_REQUEST', (state, action) => ({
  loading: true
}))

handleAction('ASYNC_TOGGLE_COLOR', (state, action) => ({
  color: state.color === 'red' ? 'blue' : 'red',
  loading: false
}))

const initialState = {
  color: 'red',
  loading: false
}

export default createReducer(initialState)
