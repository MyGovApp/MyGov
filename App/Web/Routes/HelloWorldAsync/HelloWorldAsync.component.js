import React, { Component, PropTypes } from 'react'
import classes from './HelloWorldAsync.styles.scss'
import Loader from 'Globals/Loader'

class HelloWorld extends Component {
  constructor () {
    super()
    this.state = {}
  }
  render () {
    const { asyncToggleColor, color, loading } = this.props
    return (
      <div className={classes.container}>
        {loading ? <Loader />
        : <h2
          title='helloWorldAsync'
          onClick={asyncToggleColor}
          className={classes.text}
          style={{ color }}>
          Hello World!
        </h2>
        }
      </div>
    )
  }
}

HelloWorld.propTypes = {
  color: PropTypes.string.isRequired,
  asyncToggleColor: PropTypes.func.isRequired,
  loading: PropTypes.bool
}

module.exports = HelloWorld
