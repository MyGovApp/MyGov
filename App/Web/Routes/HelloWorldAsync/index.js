import { injectReducer } from '../../Store/reducers'

export default (store) => ({
  path: 'hello-world-async',
  getComponent (nextState, next) {
    require.ensure([
      './helloWorldAsync.container',
      '../../../Modules/HelloWorldAsync/HelloWorldAsync.modules'
    ], (require) => {
      const helloWorldAsync = require('./helloWorldAsync.container').default
      const helloWorldAsyncReducer = require('../../../Modules/HelloWorldAsync/HelloWorldAsync.modules').default

      injectReducer(store, {
        key: 'helloWorldAsync',
        reducer: helloWorldAsyncReducer
      })

      next(null, helloWorldAsync)
    })
  }
})
