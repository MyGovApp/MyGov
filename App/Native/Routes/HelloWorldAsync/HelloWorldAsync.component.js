import React, { Component, PropTypes } from 'react'
import { View, Text, ActivityIndicator } from 'react-native'
import { classes } from './HelloWorldAsync.styles'

export default class HelloWorldAsync extends Component {
  render () {
    const { asyncToggleColor, color, loading } = this.props

    return (
      <View style={classes.mainView}>
        {loading
          ? <ActivityIndicator />
          : <Text
            onPress={asyncToggleColor}
            style={[classes.helloWorld, { color }]}>
            Hello World!
          </Text>
        }
      </View>
    )
  }
}

HelloWorldAsync.propTypes = {
  color: PropTypes.string.isRequired,
  asyncToggleColor: PropTypes.func.isRequired,
  loading: PropTypes.bool
}
