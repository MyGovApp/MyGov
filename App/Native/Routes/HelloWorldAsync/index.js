import HelloWorldAsync from './HelloWorldAsync.container.js'

export default HelloWorldAsync
