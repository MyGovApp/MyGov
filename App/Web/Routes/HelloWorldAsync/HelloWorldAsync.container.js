import { connect } from 'react-redux'
import { asyncToggleColor } from '../../../Modules/HelloWorldAsync/HelloWorldAsync.modules'
import HelloWorldAsync from './HelloWorldAsync.component'

const mapDispatchToProps = {
  asyncToggleColor
}

const mapStateToProps = (state, ownProps) => {
  const { color, loading } = state.helloWorldAsync
  return { color, loading }
}

export default connect(mapStateToProps, mapDispatchToProps)(HelloWorldAsync)
