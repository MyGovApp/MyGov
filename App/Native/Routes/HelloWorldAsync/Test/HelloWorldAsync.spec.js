import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper, ContainerWrapper } from './HelloWorldAsync.setup'

describe('Native Hello World Async', () => {
  it('should render the HelloWorldAsync component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the HelloWorldAsync container correctly', () => {
    const tree = renderer.create(ContainerWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
