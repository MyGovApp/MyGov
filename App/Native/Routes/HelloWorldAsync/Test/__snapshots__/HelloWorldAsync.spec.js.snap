// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native Hello World Async should render the HelloWorldAsync component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Hello World!
  </Text>
</View>
`;

exports[`Native Hello World Async should render the HelloWorldAsync container correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Hello World!
  </Text>
</View>
`;
